pages/docs/recipes/store-values.en-US.mdx

```diff
@@ -86,6 +86,39 @@ userStore.set((state) => state.user = /* value from somewhere */);
 
 Updating the store using `set()` will update all variables subscribed to the store, including (like in the example) `user`.
 
+#### App context
+
+Provided by `@openmrs/esm-framework` it's another way to store data in O3. If you are familiar with [React's context](https://react.dev/reference/react/useContext#reference), it works similarly. Unlike React's Context, App Context works
+across React subtrees, and thus can work in extensions and workspaces. All you need is to create a global context in
+some higher level component and access it somewhere in any of its children. The `useDefineAppContext` hook takes in
+the name of your context and its initial values.
+
+#### Definition
+
+```ts
+const [dateRange, setDateRange] = useState<Date[]>([
+  dayjs().startOf("day").toDate(),
+  new Date(),
+]);
+useDefineAppContext<DateFilterContext>("laboratory-date-filter", {
+  dateRange,
+  setDateRange,
+});
+```
+
+#### Usage
+
+```ts
+const { dateRange, setDateRange } = useAppContext<DateFilterContext>(
+  "laboratory-date-filter"
+) ?? {
+  dateRange: "fallback-date-range",
+  setDateRange: "fallback setter function",
+};
+```
+
+the fallback object after the `nullish coalescing` check is for providing a fallback value when the app context is undefined.
+
```

Comment:
```suggestion
##### Accessing the context

```tsx
const { dateRange, setDateRange } = useAppContext<DateFilterContext>(
  "laboratory-date-filter"
) ?? {
  // Fallback values used when the app context is undefined
  dateRange: "fallback-date-range",
  setDateRange: "fallback setter function",
};
```

```